vendmachine/VendingMachine.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedWriter;
+import java.io.BufferedReader;
+import java.io.IOException;
+
+class VendingMachine {
+    private HashMap<String, Product> products = new HashMap<>();
+    private BufferedWriter log = null;
+    private String user = null;
+
+    private void close() {
+        try {
+            BufferedWriter state = new BufferedWriter(new FileWriter("state.csv"));
+            Iterator it = products.values().iterator();
+            while (it.hasNext()) {
+                Product p = (Product) it.next();
+                state.write(p.getName()+", "+p.getPrice()+", "+p.getDescription()+"\n");
+            }
+            state.close();
+            log.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
💡  Would be better to have proper exception handling here rather than exposing to the customers the stack trace. My recommendation would be to create a custom Vending machine exception with proper message related to issue. More about here: https://docs.oracle.com/javase/tutorial/essential/exceptions/throwing.html